doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,526 @@
+  ignored, for example ``1_000_000`` is the same as ``1000000``.
+
+  .. insertprodn numeral digit
+
+  .. prodn::
+     numeral ::= {+ @digit } {? . {+ @digit } } {? {| e | E } {? {| + | - } } {+ @digit } }
+     int ::= {? - } {+ @digit }
+     num ::= {+ @digit }
+     digit ::= 0 .. 9
+
+Strings
+  Strings begin and end with ``"`` (double quote).  Use ``""`` to
+  represent a double quote character within a string.  In the grammar,
+  strings are identified with :production:`string`.
+
+Keywords
+  The following character sequences are reserved keywords that cannot
+  be used as identifiers::
+
+    _ Axiom CoFixpoint Definition Fixpoint Hypothesis IF Parameter Prop
+    SProp Set Theorem Type Variable as at by cofix discriminated else
+    end exists exists2 fix for forall fun if in lazymatch let match
+    multimatch return then using where with
+
+  Note that notations and plugins may define additional keywords.
+
+Other tokens
+  The set of tokens defined at any given time can vary because the
+  :cmd:`Notation` command can define new tokens.  A :cmd:`Require`
+  command may load more notation definitions, while the end of a
+  :cmd:`Section` may remove notations.  Some notations are defined in
+  the standard library (see :ref:`thecoqlibrary`) and are generally
+  loaded automatically at startup time.
+
+  Here are the character sequences that |Coq| directly defines as tokens
+  without using :cmd:`Notation`::
+
+    ! #[ % & ' ( () (bfs) (dfs) ) * ** + , - ->
+    . .( .. ... / : ::= := :> :>> ; < <+ <- <:
+    <<: <= = => > >-> >= ? @ @{ [ [= ] _
+    `( `{ { {| | |- || }
+
+  When multiple tokens match the beginning of a sequence of
+  characters, the longest matching token is used.  Occasionally you
+  may need to insert spaces to separate tokens.  For example, if ``~``
+  and ``~~`` are both defined as tokens, the inputs ``~ ~`` and ``~~``
+  generate different tokens, whereas if `~~` is not defined, then the
+  two inputs are equivalent.
+
+Essential vocabulary
+--------------------
+
+This section presents the most essential notions to understand the
+rest of the |Coq| manual: :term:`terms <term>` and :term:`types
+<type>` on the one hand, :term:`commands <command>` and :term:`tactics
+<tactic>` on the other hand.
+
+.. glossary::
+
+   term
+
+     Terms are the basic expressions of |Coq|.  Terms can represent
+     mathematical expressions, propositions and proofs, but also
+     executable programs and program types.
+
+     The syntax of terms is as follows.  Each one of the listed
+     constructs is presented in a dedicated section.  Some of these
+     constructs (like :n:`@term_forall_or_fun`) are part of the core
+     language that the kernel of |Coq| understands and are therefore
+     described in :ref:`the present chapter <core-language>`, while
+     some others (like :n:`@term_if`) are language extensions that are
+     presented in :ref:`the next chapter <extensions>`.
```

Comment:
"the present chapter" and "the next chapter are placeholders?  Eventually the first should directly link to the types section, the second should have the title.  You think?

Reply:
No, they're not, it's to remind the distinction between the Core language and the Language extensions chapters that we have made in the general introduction and the introductions of these chapters.